imdclient/IMDClient.py

```diff
@@ -237,6 +249,17 @@ def _go(self):
         self._conn.sendall(go)
         logger.debug("IMDClient: Sent go packet to server")
 
+    def _wait(self, IMDwait=None):
+        """
+        Wait for the server to send a pause packet
+        """
+        if IMDwait is not None:
+            wait = create_header_bytes(IMDHeaderType.IMD_WAIT, IMDwait)
+            self._conn.sendall(wait)
```

Comment:
I think this should be wrapped in a try-except-finally block to ensure that disconnect is called afterwards. You can potentially keep this method as is and have the block inside of the _disconnect method